### ISSUE TITLE
issue with qtcreator 4.1.0

### ISSUE THREAD

**Issue by ovenpasta** [nomadbyte/qtcreator-plugin-fossil#1]
Hi, I'm trying to compile the code for qtcreator version 4.1.0, I'm using qtcreator sources from Arch Linux PKGBUILD
However:

```
In file included from /home/aldo/src/qtcreator/src/qt-creator/src/plugins/fossil/fossilcontrol.cpp:29:0:
/home/aldo/src/qtcreator/src/qt-creator/src/plugins/fossil/fossilclient.h:92:10: error: conflicting return type
 specified for ‘virtual void Fossil::Internal::FossilClient::annotate(const QString&, const QString&, const QSt
ring&, int, const QStringList&)’
     void annotate(const QString &workingDir, const QString &file,
          ^~~~~~~~
In file included from /home/aldo/src/qtcreator/src/qt-creator/src/plugins/fossil/fossilclient.h:33:0,
                 from /home/aldo/src/qtcreator/src/qt-creator/src/plugins/fossil/fossilcontrol.cpp:29:
/home/aldo/src/qtcreator/src/qt-creator/src/plugins/vcsbase/vcsbaseclient.h:165:26: error:   overriding ‘virtua
l VcsBase::VcsBaseEditorWidget* VcsBase::VcsBaseClient::annotate(const QString&, const QString&, const QString&
, int, const QStringList&)’
     VcsBaseEditorWidget *annotate(

```

**Comment by nomadbyte** [comment]
As of now the most recent release is [`plugin-fossil-4.0.1_1`](https://github.com/nomadbyte/qtcreator-plugin-fossil/releases/tag/v4.0.1_1) -- respectively it targets `QtCreator-4.0.1`, should also be compatible with `QtCreator-4.0.2` (both are part of `Qt-5.6` LTS releases).

I understand that you're trying to build the `plugin-fossil-4.0.1_1` with `QtCreator-4.1.0` sources. Not sure by intent or by mistake. `QtCreator-4.1.0` is not expected to be source-compatible with the previous versions. As you have found it in your case -- interface of the underlying `VcsBaseClient` base-class has been changed in `4.1.0`, thus the compile error.

If `QtCreator-4.1.0` is not a **required** version for you, then you may try `v4.0.1` instead. Or grab the [`qtcreator-fossil-4.0.1_1`](https://github.com/nomadbyte/qtcreator-plugin-fossil/releases/tag/qtcreator-fossil-4.0.1_1) release sources with integrated `plugin-fossil`. Otherwise, work is in progress...

**Comment by ovenpasta** [comment]
I see... I've compiled successfully with qtcreator 4.0.1_1....
Thank you :)

**Comment by nomadbyte** [comment]
Adapted to `QtCreator-4.1.0`, commit: [57420b18a](https://github.com/nomadbyte/qtcreator-plugin-fossil/commit/57420b18ae1735f94e66e345ee0169c4de447960).
Still on `develop` branch and un-tagged, but you may try building it the same way you first did; just grab the plugin sources off `develop`:

```
  wget --output-document qt-creator-4.1.0-fossil.tar.gz https://github.com/nomadbyte/qtcreator-plugin-fossil/archive/develop.tar.gz
```

**Comment by nomadbyte** [comment]
Released [plugin-fossil-4.1.0_1](https://github.com/nomadbyte/qtcreator-plugin-fossil/releases/tag/v4.1.0_1). Also includes a few fixes.

> **NOTE**: This release is **plugin-fossil only**, so unlike previous releases this does not include a corresponding branch with merged `QtCreator-4.1.0` source. Merge manually as described in [README](https://github.com/nomadbyte/qtcreator-plugin-fossil/blob/v4.1.0_1/README.md). 

These fixes were also back-ported into the supported versions, so if you use `v4.0.1`, you may want to update to [plugin-fossil-4.0.1_2](https://github.com/nomadbyte/qtcreator-plugin-fossil/releases/tag/v4.0.1_2) -- that one has the corresponding merged `QtCreator-4.0.1` branch updated as well at [qtcreator-fossil-4.0.1_2](https://github.com/nomadbyte/qtcreator-plugin-fossil/releases/tag/qtcreator-fossil-4.0.1_2).
